### PR TITLE
Optim-wip: Add Neural Style Transfer tutorial & pre-trained VGG-16 model from Caffe

### PR DESCRIPTION
**Pull request by ProGamerGov**
* Added the pretrained VGG-16 model from Caffe used by the Visual Geometry Group for the ImageNet ILSVRC-2014 Challenge.
* Added neural style transfer (nst) tutorial notebook for InceptionV1 and VGG models.

~~The InceptionV1 portion of the tutorial may fail until the input range changes from https://github.com/pytorch/captum/pull/655 are implemented.~~ I updated the transform_input function in this PR to match the changes made in the other PR, so https://github.com/pytorch/captum/pull/655 is no longer required for this PR.

### CODE REVIEW

**Comment by ProGamerGov** [comment]
Some of the style images in the tutorial are currently failing to download as Wikipedia is now requiring we supply a header with a download request.

**Comment by aobo-y** [comment]
Hi @ProGamerGov, learned a lot by goring through the tutorial! But do you mind providing me more context? What is the goal and why do you believe Captum needs a neural style transfer tutorial? I can see it demonstrates how to leverage some newly added utility & loss functions to redo a similar PyTorch tutorial. However, why can't we show them in a tutorial with stronger taste of model interpretability?

**Comment by ProGamerGov** [comment]
Hi @aobo-y, thank you for reviewing this tutorial and it's associated code!

In terms of the benefits of including an NST tutorial, I have the following:

- Show users how to use a custom optimization function for optimizers that require closure functions. No other tutorial is able to cover this at the moment.
- Illustrate the usage and benefits of FFT image parameterization and transform robustness and show how in some cases they may not be the optimal solution. Currently other optimization tasks we showcase only the benefits of using FFT image parameterization and transform robustness, and ignore that model architecture has a strong influence on the effectiveness of these areas. We don't currently have any other readily available models that showcase this aspect.
- Show users how to work with custom input ranges.
- Demonstrate the use of `ImageTensor`’s `open()` class function to load images.
- Lucid’s NST tutorial is seemingly one of it’s most popular tutorials and thus having a Captum equivalent will aid in user migrations.

**Comment by ProGamerGov** [comment]
Hi @aobo-y, thank you for reviewing this tutorial and it's associated code!

In terms of the benefits of including an NST tutorial, I have the following:

- Show users how to use a custom optimization function for optimizers that require closure functions. No other tutorial is able to cover this at the moment.
- Illustrate the usage and benefits of FFT image parameterization and transform robustness and show how in some cases they may not be the optimal solution. Currently other optimization tasks we showcase only the benefits of using FFT image parameterization and transform robustness, and ignore that model architecture has a strong influence on the effectiveness of these areas. We don't currently have any other readily available models that showcase this aspect.
- Show users how to work with custom input ranges.
- Demonstrate the use of `ImageTensor`’s `open()` class function to load images.
- Lucid’s NST tutorial is seemingly one of it’s most popular tutorials and thus having a Captum equivalent will aid in user migrations.

**Comment by ProGamerGov** [comment]
We can probably skip this PR for now.
